### ISSUE TITLE
Error: "Filename STIN doesn't match the expected filename 'stdin'"

### ISSUE THREAD

**Issue by rdebeasi** [AtomLinter/linter-phpcs#60]
With the latest version of linter and linter-phpcs, I get the following error above any "real" errors that the linter detects:

`Error: "Filename STIN doesn't match the expected filename 'stdin' at line 1 col 1`

This happens when I use the "WordPress-VIP" standard that I've installed locally, but not when I use the default standard. Running `phpcs --report=full --standard=WordPress-VIP "path/myfile.php"` does not produce any errors.

Running `pear upgrade-all` does not fix the problem.

Any idea what might be happening here? Thank you!

**Comment by steelbrain** [comment]
cc @Arcanemagus @skyrpex

**Comment by skyrpex** [comment]
@rdebeasi, could you please run the following command and print the output here?

```cat "path/myfile.php" | phpcs --report=json --standard=WordPress-VIP```

**Comment by rdebeasi** [comment]
Pretty printed for readability:

```
{
    "totals": {
        "errors": 1,
        "warnings": 0,
        "fixable": 0
    },
    "files": {
        "STDIN": {
            "errors": 1,
            "warnings": 0,
            "messages": [
                {
                    "message": "Filename \"STDIN\" doesn't match the expected filename \"stdin\"",
                    "source": "Generic.Files.LowercasedFilename.NotFound",
                    "severity": 5,
                    "type": "ERROR",
                    "line": 1,
                    "column": 1,
                    "fixable": false
                }
            ]
        }
    }
}
```

I intentionally introduced a style violation and then ran the command again:
```
{
    "totals": {
        "errors": 3,
        "warnings": 0,
        "fixable": 2
    },
    "files": {
        "STDIN": {
            "errors": 3,
            "warnings": 0,
            "messages": [
                {
                    "message": "Filename \"STDIN\" doesn't match the expected filename \"stdin\"",
                    "source": "Generic.Files.LowercasedFilename.NotFound",
                    "severity": 5,
                    "type": "ERROR",
                    "line": 1,
                    "column": 1,
                    "fixable": false
                },
                {
                    "message": "Expected 1 spaces after opening bracket; 0 found",
                    "source": "PEAR.Functions.FunctionCallSignature.SpaceAfterOpenBracket",
                    "severity": 5,
                    "type": "ERROR",
                    "line": 41,
                    "column": 32,
                    "fixable": true
                },
                {
                    "message": "Expected 1 spaces before closing bracket; 0 found",
                    "source": "PEAR.Functions.FunctionCallSignature.SpaceBeforeCloseBracket",
                    "severity": 5,
                    "type": "ERROR",
                    "line": 41,
                    "column": 32,
                    "fixable": true
                }
            ]
        }
    }
}
```

The error also appears with the "full" report. Running the same phpcs command without `cat`does not produce the "Filename STDIN" error.

I have phpcs installed through PEAR, for what it's worth.

**Comment by skyrpex** [comment]
Thanks for the pretty print.

Both the linter and PHPCS itself seems to work fine. It's just that your standard (WordPress-VIP) expects your PHP filenames to be lowercase. PHPCS seems to assume that STDIN is a filename...

Maybe we should file a bug at PHPCS. What do you think, @steelbrain?

**Comment by steelbrain** [comment]
@skyrpex Instead of taking the long way around, I would suggest that we handle both the cases where it's lowercase and where it's uppercase in this package. Opening a bug won't hurt anyone, so opening it should get it fixed in the future.

**Comment by skyrpex** [comment]
So, shall we ignore ```Generic.Files.LowercasedFilename.NotFound``` errors from PHPCS until it gets fixed?

**Comment by steelbrain** [comment]
I kinda misread the text above :sweat_smile:  but yeah. Ignore that error.

**Comment by rdebeasi** [comment]
Awesome, thank you so much! I added this change to my local copy of the plugin, and it resolves the problem.

**Comment by skyrpex** [comment]
Nice! Is it still showing other messages?

**Comment by rdebeasi** [comment]
Yep, those appear as expected!

**Comment by Arcanemagus** [comment]
@rdebeasi what version of `phpcs` are you running?

After further investigation it appears this issue should have been [fixed already](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Generic/Sniffs/Files/LowercasedFilenameSniff.php#L54-L56) at the source, which would explain why I haven't been seeing it in 2.3.4.

**Comment by Arcanemagus** [comment]
Nevermind, the fix was in there because @gsherwood _just_ implemented it. I still have no idea why I couldn't reproduce this initially.
